<a href="https://colab.research.google.com/github/bhusasravanii/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/bhusasravanii/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [20]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 4))
print("Average Search Volume:", round(df["search_volume"].mean(), 2))

Number of rows: 30000
Number of columns: 44

Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.5107
Average Search Volume: 158.88


## Quick Look at the Data

I explored the starter dataset to understand whether it is suitable for my research question.

Observations:
- The dataset contains **30,000 webpages** and **44 features**, providing enough data for analysis.
- Most pages are labeled as **down (16,262)**, followed by **stable (5,962)** and **up (4,388)**, showing that the dataset contains a variety of page performance trends.
- The **average CTR is 0.5107**, and the **average search volume is 158.88**, indicating variation in search performance across pages.

These observations suggest that the dataset is suitable for investigating which webpages should be prioritized for content refresh.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

**Content Refresh Prioritization**

I chose the Content Refresh Prioritization lane because many websites contain pages that gradually lose traffic and need to be updated. My research question is how machine learning can help identify which webpages should be refreshed first. This can help SEO teams save time by prioritizing pages that are most likely to benefit from content updates.

## My Lane

**Content Refresh Prioritization**

I chose the Content Refresh Prioritization lane because many websites contain pages that gradually lose traffic and need to be updated. My research question is how machine learning can help identify which webpages should be refreshed first. This can help SEO teams prioritize pages that need updates and improve search performance.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question

### Decision
Determine which webpages should be prioritized for content refresh.

### Action
The SEO or content team will update the webpages predicted to benefit the most from content improvements.

### Cost of a Wrong Recommendation
If the recommendation is incorrect, important pages may be ignored while time and resources are spent updating pages that do not need changes. This can reduce the effectiveness of the content refresh strategy.

### Why Machine Learning?
Machine learning can analyze many webpage features simultaneously and identify patterns that are difficult to detect using simple manual rules.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [21]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 4))
print("Average Search Volume:", round(df["search_volume"].mean(), 2))

Number of rows: 30000
Number of columns: 44

Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.5107
Average Search Volume: 158.88


In [22]:
import os

print("Current directory:", os.getcwd())
print("Files here:")
print(os.listdir())

Current directory: /content/flyrank-ml-internship/flyrank-ml-internship
Files here:
['notebooks', 'data', 'requirements.txt', 'README.md', 'skills', '.gitignore', 'scripts', 'CLAUDE.md', 'DATA_USE.md', '.git', 'submission', 'work', 'AGENTS.md', 'outputs', '.github', 'LICENSE', 'docs', 'SETUP.md', '01_first_look_and_discovery.ipynb', 'GUIDE.md']


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project is intended to identify webpages that may benefit from content refresh based on patterns in historical search performance data.

The model does not prove that updating a webpage will increase traffic or improve Google rankings. It only provides recommendations based on the available data.

The results should be used as decision support for SEO teams rather than as automatic decisions. Human review is still necessary before taking action.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## Self-Check

- ✅ I selected one research lane.
- ✅ I explained the decision, action, and cost of a wrong recommendation.
- ✅ I supported my idea using real numbers from the starter dataset.
- ✅ I used careful language and avoided making unsupported claims.
- ✅ My notebook is ready for future improvements during the internship.

## Self Check

✔ I selected one research lane.

✔ I clearly defined the decision, action, and cost of a wrong recommendation.

✔ I supported my research question using real numbers from the dataset.

✔ I used careful language and avoided making claims that cannot be proven.

✔ I understand that this is a framing exercise, not a model-building exercise.